src/Grpc.Net.Client/Balancer/PickContext.cs

```diff
@@ -0,0 +1,38 @@
+﻿#region Copyright notice and license
+
+// Copyright 2019 The gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#endregion
+
+using System.Net.Http;
+
+namespace Grpc.Net.Client.Balancer
+{
+    /// <summary>
+    /// Context used to pick a <see cref="Subchannel"/>.
+    /// </summary>
+#if HAVE_LOAD_BALANCING
+    public
+#else
+    internal
+#endif
+        sealed class PickContext
+    {
+        /// <summary>
+        /// Gets or sets the <see cref="HttpRequestMessage"/>.
+        /// </summary>
+        public HttpRequestMessage? Request { get; set; }
```

Comment:
I made it a property because more arguments might get added here in the future. We could end up with many constructors. Setting properties seems easier.